src/parts/Lumina/index.jsx

```diff
@@ -0,0 +1,25 @@
+import { useRef, useState, useEffect, createContext } from "react";
+
+import { spawnNode } from "lumina-node";
+
+export const LuminaContext = createContext(null);
+
+export function LuminaContextProvider({ children }) {
+  const [lumina, setLumina] = useState(null);
+  const initialized = useRef(false);
```

Comment:
can we switch to `useState` here? I overused useRef I think :)

Reply:
I'm not sure how to make it not initialise worker twice (that's what's happening with naively replacing ref with state here) :sweat_smile: 
I'll tinker a bit more